crates/data_components/src/token_wrapper.rs

```diff
@@ -0,0 +1,54 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use snafu::prelude::*;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("Unable to get token: {source}"))]
+    UnableToGetToken {
+        source: Box<dyn std::error::Error + Send + Sync>,
+    },
+
+    #[snafu(display("Unable to refresh token: {source}"))]
+    UnableToRefreshToken {
+        source: Box<dyn std::error::Error + Send + Sync>,
+    },
+}
+
+pub type Result<T, E = Error> = std::result::Result<T, E>;
+
+#[async_trait]
+pub trait TokenWrapper: Send + Sync {
+    fn is_refreshable(&self) -> bool;
```

Comment:
Is this function needed?

Reply:
I would expect the code that consumes this would just call `get_token` and it would handle the details of refreshing if needed.

Reply:
No, not anymore, I forgot to remove